hal/include/HAL/handles/IndexedHandleResource.h

```diff
@@ -40,31 +40,23 @@ class IndexedHandleResource {
   IndexedHandleResource(const IndexedHandleResource&) = delete;
   IndexedHandleResource operator=(const IndexedHandleResource&) = delete;
   IndexedHandleResource();
-  ~IndexedHandleResource();
+  ~IndexedHandleResource() = default;
```

Comment:
Shouldn't need to actually explicitly make this `= default`, just remove it entirely.